### PR TITLE
Vapor 3

### PR DESCRIPTION
**Pull request by BrettRToomey**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -2,41 +2,41 @@ version: 2
...
             brew install cmysql
             brew install ctls
-            brew install cstack
+            brew install libressl
       - run:
           name: Build and Run Tests
           no_output_timeout: 1800
           command: |
-            swift package generate-xcodeproj --enable-code-coverage
+            swift package generate-xcodeproj --enable-code-coverage
```
Comment (steffendsommer) [inline .circleci/config.yml]: The trailing spaces can be removed

File: Package.swift [hunk 2]
```diff
@@ -1,9 +1,23 @@
+// swift-tools-version:4.0
```
Comment (steffendsommer) [inline Package.swift]: This should be 4.1 to be precise

File: README.md [hunk 3]
```diff
@@ -1,154 +1,109 @@
...
-apt-get install cstack
-```
-
 ### Integrating Bugsnag in your project
 
 Update your `Package.swift` file.
 
 ```swift
-.Package(url: "https://github.com/nodes-vapor/bugsnag.git", majorVersion: 2)
+// .Package(url:"https://github.com/nodes-vapor/bugsnag.git", majorVersion: 3)
```
Comment (steffendsommer) [inline README.md]: This can be removed

File: Sources/Bugsnag/Breadcrumb.swift [hunk 4]
```diff
@@ -0,0 +1,59 @@
+import Vapor
+
+final class BreadcrumbContainer: Service {
+    var breadcrumbs: [BugsnagBreadcrumb] = []
+
+    init() {}
+}
```
Comment (steffendsommer) [inline Sources/Bugsnag/Breadcrumb.swift:7]: Should we have blank newline here to separate the class and enum?

File: Sources/Bugsnag/BugsnagReporter.swift [hunk 5]
```diff
@@ -0,0 +1,306 @@
...
+            let body = try buildBody(
+                request,
+                error: error,
+                severity: severity,
+                userId: userId,
+                userMetadata: metadata,
+                callsite: callsite
+            )
+
+            return HTTPClient.connect(hostname: "notify.bugsnag.com", on: request)
```
Comment (steffendsommer) [inline Sources/Bugsnag/BugsnagReporter.swift]: `hostName` has been defined at the top (needs to be updated slightly to exclude scheme)

File: Sources/Bugsnag/BugsnagReporter.swift [hunk 6]
```diff
@@ -0,0 +1,306 @@
...
+public final class BugsnagReporter: Service, Middleware {
+    public let apiKey: String
+    public let releaseStage: String
+    public let debug: Bool
+
+    let hostName = "https://notify.bugsnag.com/"
+    let payloadVersion: UInt8 = 4
+    let app: BugsnagApp
+
+    var userTypes: [Authenticatable.Type] = []
```
Comment (steffendsommer) [inline Sources/Bugsnag/BugsnagReporter.swift]: This is not used

**Review by steffendsommer — DISMISSED** [review]
Good stuff! A couple of minor comments and then I think we need one more common/critical feature: the configuration of a whitelist for release stages/environments. A common case for us is to not report while testing locally.

**Review by siemensikkema — APPROVED** [review]

**Pull request merged**